libraries/core_libs/network/src/tarcap/packets_handlers/common/ext_votes_packet_handler.cpp

```diff
@@ -9,24 +9,29 @@ ExtVotesPacketHandler::ExtVotesPacketHandler(std::shared_ptr<PeersState> peers_s
                                              const std::string &log_channel_name)
     : PacketHandler(std::move(peers_state), std::move(packets_stats), node_addr, log_channel_name) {}
 
-void ExtVotesPacketHandler::onNewPbftVote(std::shared_ptr<Vote> &&vote) {
+void ExtVotesPacketHandler::onNewPbftVote(std::shared_ptr<Vote> &&vote, bool reward_vote) {
   std::vector<dev::p2p::NodeID> peers_to_send;
   const auto round = vote->getRound();
   for (auto const &peer : peers_state_->getAllPeers()) {
-    if (!peer.second->isVoteKnown(vote->getHash()) && !peer.second->syncing_ && peer.second->pbft_round_ <= round) {
+    if (!peer.second->isVoteKnown(vote->getHash()) && !peer.second->syncing_ &&
+        (peer.second->pbft_round_ <= round || reward_vote)) {
```

Comment:
is this ok ? Even if it is reward vote, difference between peer's round and vote round should be max 1

Reply:
I may misunderstand your question. This is not checking "difference between peer's round and vote round should be max 1".
This is "or" operator. If "it's reward vote" OR "peer round <= vote round"

Reply:
yes you did. I mean - even if it is a reward vote, you should check difference between peer's round. Whay sense does it make to send reward vote with round 10 if the peer's pbft_round is already 12 ? That means he must have already  processed block 11, which pointed to the reward votes from round 10.

Reply:
No, difference between peer's round and reward votes round should be any number
nodes don't finalize blocks in every PBFT round